### DISCUSSION TITLE
calling simple scripts with args from the command line?

### DISCUSSION THREAD

**Discussion by cfritze64** [johnlindquist/kit#38] [Ideas]
really enjoying playing with this tool. Many thanks.

Any provision for passing args to a script via the CLI? I can put args in a file and read that within the script, but a direct call would be efficient...

**Comment by johnlindquist** [comment]
It should "just work" if you pass an arg. The arg prompts should only appear if the arg isn't provided.

Maybe I'm misunderstanding your question?

**Comment by cfritze64** [comment]
>>Maybe I'm misunderstanding your question?

Or just as likely, as a newbie, my language discussing these things is probably not very clear.

Say I have a script named find-president where the first line of code is something like:

> let theQuery = await arg('Search for last name:');

this prompts the user to enter a search term, e.g. "Washington"
What if I wanted a script that instead of interactively asking for a name, accepts one from the CLI call, something like:

> simple run find-president Washington

or some such. This would let me string together steps within and outside simple scripts, passing in arguments from the command line.

**Reply by johnlindquist** [reply]
Yes, that should "just work".

Here's what I recommend instead:

```
~/.simple/bin/find-president Washington
```

## Why does this work?
You'll see bash scripts are generated inside the `~/.simple/bin` directory whenever you run the `new` command.

If you add `~/.simple/bin` to your $PATH, your terminal will know about the `~/simple/bin` dir then you can just run `find-president Washington`


## Adding ~/.simple/bin to your $PATH:
> The following command will attempt to add `~/.simple/bin` to your path. (this was run automatically if you used the install script). You'll have to restart your terminal for it to take effect:
```
SIMPLE_PATH=~/.simple && SIMPLE_PROFILE=$($SIMPLE_PATH/config/detect-profile.sh) $SIMPLE_PATH/config/link-profile.sh
```

Does that help?

**Reply by cfritze64** [reply]
Yes. Absolutely gives me what I was looking for. thanks for the thorough explanation.
